### ISSUE TITLE
Module isn't working

### ISSUE THREAD

**Issue by maverick85** [openprovider/OP-WHMCS7#83]
Hi,

I came about your company recently and decided to give it a go on my whmcs store.
I downloaded the module from [OpenProvider - Our Plugins](https://openprovider.com/services/our-plugins/).
Followed the instructions.

Two issues:
1. Does not authenticate;
2. Breaks WHMCS Admin Dashboard;

Setup:
1. I activate the plugin according to instructions, set the URL to https://api.openprovider.eu (also tested with https://api.openprovider.eu/v1beta/ - same result)
2. I insert the user with API access enabled and the password (not the hash, the password!) 

I keep getting the error: `Login Unseccessful: please ensure credentials and URL are correct`

![Screenshot 2020-05-14 at 19 34 24](https://user-images.githubusercontent.com/10200278/81972725-ec89ee80-961a-11ea-9b82-8313d9a24ecb.png)
Also going back to WHMCS Admin Dashboard, it displays a blank slate:
![Screenshot 2020-05-14 at 19 32 46](https://user-images.githubusercontent.com/10200278/81972855-2529c800-961b-11ea-87d8-aafec0c05c8b.png)
The log:
![Screenshot 2020-05-14 at 19 33 39](https://user-images.githubusercontent.com/10200278/81972918-438fc380-961b-11ea-8af5-8c26a764c594.png)
As soon as I disable OpenProvider module, the admin dashboard works ok.

I have also whitelisted my server's main IP and the dedicated IP of the website on the OpenProvider control panel.

```
PHP 7.3.17 (cli) (built: May  5 2020 19:27:48) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.17, Copyright (c) 1998-2018 Zend Technologies
    with the ionCube PHP Loader + ionCube24 v10.3.9, Copyright (c) 2002-2019, by ionCube Ltd.
    with Zend OPcache v7.3.17, Copyright (c) 1999-2018, by Zend Technologies
```

From OpenProvider [WHMCS Frequently Asked Questions](https://support.openprovider.eu/hc/en-us/articles/360009201193):

> 2. Why am I getting an authentication error?
Check the following things, after changing your password in Openprovider:
Ensure that the correct password is being entered into the module
Ensure that the password, and not the hash, is entered in the module
Ensure that two-factor authentication is not enabled for the credentials which you are using for the module. (To maintain security, create a technical contact in your Reseller Control Panel, and whitelist only the IP addresses of your WHMCS instance )
Ensure that you see that the module authenticates after clicking "save" (see screenshot, the DNS template box will appear)

-> The correct password is being entered;
-> The password, not the hash, was entered;
-> tfa is not enabled, I am using an user added as Technical Contact with API Access enabled
-> The module does not authenticate.

I had no IP's whitelisted and for a second I thought "maybe I do need to put an IP there" - according to the documentation it didn't seem like it, but anyway.
Was not working before, not working after. 

I believe this is a critical issue as this module simply DOES NOT WORK and breaks WHMCS.

**Comment by wmetge** [comment]
Hi @maverick85, 

Thank you for taking the time to post an issue. We're sorry to hear you're having difficulty with the module, and if you contact Openprovider support directly, they'll be happy to individually investigate the issue you're experiencing. 

If you provide your reseller ID, our support can look at what error is being sent when the module attempts to login. 

From what you've said, it seems everything is configured correctly, however we do have quite a few users successfully using current and past versions of the model, so it makes me suspect that there is something particular about your experience.

if you provide logs from your server (either here or to our support team) from when you get the blank dashboard, it might provide insight into the cause. We certainly will fix any issues where the module causes problems to the WHMCS instance itself.

**Comment by maverick85** [comment]
Hi @wmetge thank you for the reply.
I actually did contact Openprovider directly this morning and also opened a ticket yesterday (57512) it has my reseller ID and more info etc.

Regarding the admin dashboard issue, it was told me by Hicham that it happens because the module is not attaining authentication so the dashboard module fails and that causes the blank screen. He mentioned when authentication proceeds correctly, that behaviour will stop.

I was told there would be a team meeting today around 11am CEST and this issue would be brought up. As I mentioned to Hicham I'm taking the day just to tackle this issue.

About the dashboard, I think there should be some sort of protection if the module isn't authenticated then just disable the admin dashboard widget, but right now I'd just like to be able to authenticate. Thanks.

**Comment by wmetge** [comment]
@maverick85 Yes, I'm in contact with Hicham regarding your ticket. 

Please provide the server logs when you attempt to authenticate. In our staging environment we do not get such an error when the module is unable to connect with Openprovider, or incorrect credentials are used.

**Comment by maverick85** [comment]
Hi @wmetge thank you for the reply.

So here's where I stand:
The original password for user to setup on WHMCS had 31 characters, including percentiles, dollar signs, equals, upper and lower case letters and numbers.

So I created another technical user with API access and a simpler password.
I was able to authenticate straight away with this new user.
So I changed the password for the original user to a simpler one as well. But it did not authenticate.
I removed all IP's from the whitelist.

I was then able to authenticate.
I added the website IP to the whitelist again. It is working now.

**Comment by MrKoopie** [comment]
@maverick85 I have to assume this was API related.

In regard to the blank dashboard: I have been able to reproduce this and it happens indeed with incorrect logins. This will get fixed in 3.4.

**Comment by maverick85** [comment]
@MrKoopie thank you for your message and I'm sorry for the delay in my reply.
About the original issue, I got thinking if it could be some issue regarding the characters I used on the password. Despite very infrequently, some sites/password fields can't handle some special characters. But anyway it got solved, so far I'm happy about it and I thank you all for the support!
